MkChrootTree.sh

```diff
@@ -26,6 +27,99 @@ function err_out() {
    exit "${1}"
 }
 
+# Allow partition-string input to be arbitrarily ordered
+function SortDiskGeomString {
+   local COUNT
+   local LSTELEM
+   local MOUNTS
+   local MTELEM
+   local MTSSORTED
+   local PARTITIONARRAY
+   local PARTITIONSTR
+
+   COUNT=0
+   MOUNTS=()
+   PARTITIONSTR="${GEOMETRYSTRING}"
+   SORTEDARRAY=()
+
+   # Convert ${PARTITIONSTR} to iterable partition-info array
+   IFS=',' read -r -a PARTITIONARRAY <<< "${PARTITIONSTR}"
+   unset IFS
+
+   # Extract iterable-list of mountpoints from partition-info array
+   while [[ ${COUNT} -lt ${#PARTITIONARRAY[*]} ]]
+   do
+      MOUNTS+=( "${PARTITIONARRAY[${COUNT}]//:*/}" )
```

Comment:
To cleanup the double-loop below, how about instead of a list, use an associative array, so you end up with `<MOUNT> = <PARTITIONITEM>`? totally untested, but something like this...

```
declare -A MOUNTS
while [[ ${COUNT} -lt ${#PARTITIONARRAY[*]} ]]
do
  MOUNTS["${PARTITIONARRAY[${COUNT}]//:*/}"] = "${PARTITIONARRAY[${COUNT}]}"
done
```

then sort MOUNTS by key, and then as you loop over MOUNTS you have the LSTELEM already...

Reply:
I'd considered trying to create an associative array, but, using one array to filter another came more-quickly to mind. I'll have to cogitate on doing it now or punting til AMIgen8.

Reply:
Addressed in commit 8ada807. Appears to also shorten effected script by two-dozen+ lines.